secboot/secboot_tpm.go

```diff
@@ -191,7 +211,7 @@ func UnlockVolumeIfEncrypted(name string, lockKeysOnFinish bool) (string, error)
 			return nil
 		}
 
-		if tpmErr != nil {
+		if !tpmDeviceAvailable {
 			return fmt.Errorf("cannot unlock encrypted device %q: %v", name, tpmErr)
```

Comment:
In this scenario, it might be worth calling https://godoc.org/github.com/snapcore/secboot#ActivateVolumeWithRecoveryKey before failing with an error.

Reply:
That makes sense, will try to unlock with the recovery key in this case.